### ISSUE TITLE
Problem when using podman

### ISSUE THREAD

**Issue by mtcolman** [owenrumney/lazytrivy#31]
**Describe the bug**
Hi, firstly this looks like it will be a great tool, so thanks for creating. Secondly, this might not necessarily be a bug, because I am likely trying something that might not be expected to work - but it would be great if it did, or if you can explain why it doesn't please? In the prerequisites, its states:

> Prerequisites
>
> In order for lazytrivy to be cross-platform, it uses the Trivy docker image. This means that you will need to have Docker running on your machine for lazytrivy to work.

I use Podman instead of Docker, which from use in the past I know works fine with Trivy itself. I therefore wanted to try lazytrivy with podman. I've tried it as both non-root and root user on my Linux instance and both have had problems.

Non-root podman I get an error and cannot view the "image" UI. When running as root, the UI appears, but the output gets "mashed up".

**To Reproduce**

***Non-root:***
```
$ mkdir -p /tmp/podman-run-1000/podman
$ podman system service -t 3600 unix:///tmp/podman-run-1000/podman/podman.sock &
$ export DOCKER_HOST=unix:///tmp/podman-run-1000/podman/podman.sock
[1] 5080
$ lazytrivy --docker-host $DOCKER_HOST image
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/owenrumney/lazytrivy/pkg/dockerClient.(*Client).ListImages(0xc0004640f0)
        /home/runner/work/lazytrivy/lazytrivy/pkg/dockerClient/images.go:29 +0x53e
github.com/owenrumney/lazytrivy/pkg/controllers/vulnerabilities.(*Controller).RefreshImages(0xc000394680)
        /home/runner/work/lazytrivy/lazytrivy/pkg/controllers/vulnerabilities/image.go:130 +0x85
github.com/owenrumney/lazytrivy/pkg/controllers/vulnerabilities.(*Controller).Initialise.func1(0xc0004620f0)
        /home/runner/work/lazytrivy/lazytrivy/pkg/controllers/vulnerabilities/vulnerability.go:39 +0x48
github.com/awesome-gocui/gocui.(*Gui).consumeevents(0xc0004620f0)
        /home/runner/go/pkg/mod/github.com/awesome-gocui/gocui@v1.1.0/gui.go:540 +0x14b
github.com/awesome-gocui/gocui.(*Gui).MainLoop(0xc0004620f0)
        /home/runner/go/pkg/mod/github.com/awesome-gocui/gocui@v1.1.0/gui.go:515 +0x247
github.com/owenrumney/lazytrivy/pkg/controllers/gui.(*Controller).Run(0xc000464140)
        /home/runner/work/lazytrivy/lazytrivy/pkg/controllers/gui/gui.go:90 +0x2c
github.com/owenrumney/lazytrivy/internal/cmd.startGUI({0xc1201c, 0xf})
        /home/runner/work/lazytrivy/lazytrivy/internal/cmd/start.go:61 +0x1dc
github.com/owenrumney/lazytrivy/internal/cmd.glob..func1(0x11adb20?, {0xc0ae41?, 0x2?, 0x2?})
        /home/runner/work/lazytrivy/lazytrivy/internal/cmd/root.go:12 +0x25
github.com/spf13/cobra.(*Command).execute(0x11adb20, {0xc0004161c0, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc00041a000)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
        /home/runner/work/lazytrivy/lazytrivy/main.go:21 +0xf8
```

***root:***
```
$ sudo -i
mkdir -p /run/podman
podman system service -t 3600 unix:///run/podman/podman.sock &
[1] 4728
export DOCKER_HOST=unix:///run/podman/podman.sock
lazytrivy --docker-host $DOCKER_HOST image
```

![image](https://user-images.githubusercontent.com/33348891/191727085-30948a4a-d949-4a3a-bbed-8453002e42b9.png)

then run scan:

![image](https://user-images.githubusercontent.com/33348891/191727195-64c3a27d-9344-41e1-b294-cadf5d306e18.png)

**Expected behavior**
I would expect the image scan to complete and be displayed as per the lazytrivy readme images.

**Screenshots**
(See above)

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04 running on WSL2 on Windows 11
 
***Additional:***
```
$ podman version
Version:      3.4.4
API Version:  3.4.4
Go Version:   go1.17.3
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```

**Comment by owenrumney** [comment]
Hey @mtcolman thanks for taking the time to raise an issue. I'll get podman set up and try to reproduce

**Comment by owenrumney** [comment]
Hey @mtcolman, managed to reproduce the noisy output - this is because podman is being started in the background and when a pull is done (aquasec/trivy:latest is pulled as part of the scanning process on the first run), podman is outputting to the terminal which is colliding with lazytrivy

I resolved this by dumping podman logs to dev/null or could be achieved by running lazytrivy in a different terminal to the one used to start podman

```
podman system service -t 3600 unix:///tmp/podman-run-1000/podman/podman.sock 2>/dev/null  &
```

As far as the error when starting in non-root, I couldn't work out what image could possibly not have a valid repotag since even an un-named one has <none>:<none>. I've been more defensive on this - if get a chance to test a commit release that would be great

```
go install github.com/owenrumney/lazytrivy@c167c92d542f3327904a68db896643fd154cde84
```

**Comment by mtcolman** [comment]
@owenrumney I've done the following, all in non-root mode. I've cleared down the local podman repo and just pulled two images:
 - alpine:3.16.1
 - hello-world (leaving tag blank, to get latest)
 
```
go install github.com/owenrumney/lazytrivy@c167c92d542f3327904a68db896643fd154cde84
mkdir -p /tmp/podman-run-1000/podman
podman system service -t 3600 unix:///tmp/podman-run-1000/podman/podman.sock 2>/dev/null &
export DOCKER_HOST=unix:///tmp/podman-run-1000/podman/podman.sock
lazytrivy --docker-host $DOCKER_HOST image
```

It shows:

![image](https://user-images.githubusercontent.com/33348891/191810429-f5f96989-3a48-4d95-8cc4-521dc843c5b9.png)

and the scan works nicely.

I then wanted to test if I overwrote a container image and left one with repository and tag = `<none>`:
```
podman build -t docker.io/library/hello-world:latest -f Dockerfile ./
```

Leaves me with:
```
podman image ls
REPOSITORY                     TAG         IMAGE ID      CREATED        SIZE
docker.io/library/hello-world  latest      6b7c43821c88  6 seconds ago  128 MB
docker.io/aquasec/trivy        latest      a99a781eca1f  6 days ago     206 MB
docker.io/library/ubuntu       20.04       a0ce5a295b63  2 weeks ago    75.2 MB
docker.io/library/alpine       3.16.1      d7d3d98c851f  2 months ago   5.82 MB
<none>                         <none>      feb5d9fea6a5  12 months ago  19.9 kB
```

And if I run lazytrivy:

![image](https://user-images.githubusercontent.com/33348891/191811129-37a16773-fd92-42ee-b066-cdb9a4125151.png)

I then removed the lazytrivy version above and installed the "latest" version and I got:
```
lazytrivy --docker-host $DOCKER_HOST image
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/owenrumney/lazytrivy/pkg/dockerClient.(*Client).ListImages(0xc00049e0f0)
        /home/matt/go/pkg/mod/github.com/owenrumney/lazytrivy@v1.0.0/pkg/dockerClient/images.go:29 +0x53e
github.com/owenrumney/lazytrivy/pkg/controllers/vulnerabilities.(*Controller).RefreshImages(0xc00007c1c0)
        /home/matt/go/pkg/mod/github.com/owenrumney/lazytrivy@v1.0.0/pkg/controllers/vulnerabilities/image.go:130 +0x85
github.com/owenrumney/lazytrivy/pkg/controllers/vulnerabilities.(*Controller).Initialise.func1(0xc0005440f0)
        /home/matt/go/pkg/mod/github.com/owenrumney/lazytrivy@v1.0.0/pkg/controllers/vulnerabilities/vulnerability.go:39 +0x48
github.com/awesome-gocui/gocui.(*Gui).MainLoop(0xc0005440f0)
        /home/matt/go/pkg/mod/github.com/awesome-gocui/gocui@v1.1.0/gui.go:508 +0x231
github.com/owenrumney/lazytrivy/pkg/controllers/gui.(*Controller).Run(0xc00049e140)
        /home/matt/go/pkg/mod/github.com/owenrumney/lazytrivy@v1.0.0/pkg/controllers/gui/gui.go:90 +0x2c
github.com/owenrumney/lazytrivy/internal/cmd.startGUI({0xc178ea, 0xf})
        /home/matt/go/pkg/mod/github.com/owenrumney/lazytrivy@v1.0.0/internal/cmd/start.go:61 +0x1dc
github.com/owenrumney/lazytrivy/internal/cmd.glob..func1(0x11b4d60?, {0xc1071d?, 0x2?, 0x2?})
        /home/matt/go/pkg/mod/github.com/owenrumney/lazytrivy@v1.0.0/internal/cmd/root.go:12 +0x25
github.com/spf13/cobra.(*Command).execute(0x11b4d60, {0xc0004a8940, 0x2, 0x2})
        /home/matt/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004eef00)
        /home/matt/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/matt/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
        /home/matt/go/pkg/mod/github.com/owenrumney/lazytrivy@v1.0.0/main.go:21 +0xf8
```

I have then double-checked by installing `go install github.com/owenrumney/lazytrivy@c167c92d542f3327904a68db896643fd154cde84` again and it can cope with `<none>` as repository or tag.

**Comment by owenrumney** [comment]
Great, so I'm understanding from that the commit version is the fix so I'll cut a release. 

Thanks for taking the time to test so thoroughly

**Comment by owenrumney** [comment]
Resolved by #32
